core-bundle/src/OptIn/OptIn.php

```diff
@@ -87,9 +84,16 @@ public function purgeTokens(): void
             return;
         }
 
+        $time = strtotime('+2 days');
         $adapter = $this->framework->getAdapter(Model::class);
 
         foreach ($tokens as $token) {
+            // Keep unconfirmed tokens for two additional days to ensure they are not removed
+            // before the cron jobs that purge unconfirmed subscriptions have run.
+            if (!$token->confirmedOn && $token->removeOn < $time) {
+                continue;
+            }
```

Comment:
This would fix the issue mentioned [here](https://github.com/contao/contao/pull/9069/files#r2590742853):

```suggestion
        $time = strtotime('-2 days');
        $adapter = $this->framework->getAdapter(Model::class);

        foreach ($tokens as $token) {
            // Keep unconfirmed tokens for two additional days to ensure they are not removed
            // before the cron jobs that purge unconfirmed subscriptions have run.
            if (!$token->confirmedOn && $token->removeOn > $time) {
                continue;
            }
```

* Member registrations are properly purged at the `removeOn` time.
* The token is kept at the `removeOn` time.
* Any unconfirmed token with a `removeOn` time older than 2 days is properly purged.
* Any expired, but confirmed tokens with an existing related record has its `removeOn` time prolonged for 3 years.